src/test/compile-fail/coercion-slice.rs

```diff
@@ -14,6 +14,5 @@ fn main() {
     let _: &[i32] = [0];
     //~^ ERROR mismatched types
     //~| expected type `&[i32]`
-    //~| found type `[{integer}; 1]`
```

Comment:
It's probably a stray side-effect? Try wrapping in `probe` if you haven't already. In fact, you probably want a version of `try_coerce` with no side-effects (call it `probe_coerce`).